openpdf/src/main/java/com/lowagie/text/pdf/PdfNumber.java

```diff
@@ -170,4 +172,25 @@ public void increment() {
         value += 1.0;
         setContent(ByteBuffer.formatDouble(value));
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        PdfNumber pdfNumber = (PdfNumber) o;
+        return Double.compare(pdfNumber.value, value) == 0;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(value);
+    }
+
+    @Override
+    public int compareTo(Object o) {
+        if (this == o) return 0;
+        if (o == null || getClass() != o.getClass()) return -1;
```

Comment:
Specify `Comparable` type and use `Objects.requireNonNull(o, <message>)`, because `null` objects must not be compared

Reply:
I agree